src/main/java/ledger/database/storage/SQLiteDatabase.java

```diff
@@ -0,0 +1,988 @@
+            "ACCOUNT_NAME TEXT           NOT NULL, " +
+            "ACCOUNT_DESC TEXT           NOT NULL" +
+            ")";
+    private static final String tableAccountBalance = "CREATE TABLE IF NOT EXISTS ACCOUNT_BALANCE " +
+            "(ABAL_ACCOUNT_ID INTEGER PRIMARY KEY AUTOINCREMENT, " +
+            "ABAL_DATETIME REAL         NOT NULL, " +
+            "ABAL_AMOUNT INT            NOT NULL " +
+            ")";
+    private static final String tablePayee = "CREATE TABLE IF NOT EXISTS PAYEE " +
+            "(PAYEE_ID INTEGER PRIMARY KEY AUTOINCREMENT, " +
+            "PAYEE_NAME TEXT           NOT NULL, " +
+            "PAYEE_DESC TEXT           NOT NULL" +
+            ")";
+    private static final String tableTagToPayee = "CREATE TABLE IF NOT EXISTS TAG_TO_PAYEE " +
+            "(TTPE_TAG_ID INT           NOT NULL, " +
+            "TTPE_PAYEE_ID INT          NOT NULL, " +
+            "FOREIGN KEY(TTPE_TAG_ID) REFERENCES TAG(TAG_ID), " +
+            "FOREIGN KEY(TTPE_PAYEE_ID) REFERENCES PAYEE(PAYEE_ID)" +
+            ")";
+
+    private Connection database;
+
+
+    public SQLiteDatabase(InputStream iStream, String pathToDb) throws StorageException {
+        // Initialize SQLite streams.
+
+        try {
+            Class.forName("org.sqlite.JDBC");
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException("Unable to find SQLite Driver", e);
+        }
+
+        try {
+            database = DriverManager.getConnection("jdbc:sqlite:" + pathToDb);
+        } catch (SQLException e) {
+            throw new StorageException("Unable to connect to JDBC Socket. ");
+        }
+        initializeDatabase();
+    }
+
+    // DB management functions
+    @Override
+    public void initializeDatabase() throws StorageException {
+        LinkedList<String> tableSQL = new LinkedList<>();
+
+        tableSQL.add(tableTag);
+        tableSQL.add(tableType);
+        tableSQL.add(tableAccount);
+        tableSQL.add(tableAccountBalance);
+        tableSQL.add(tablePayee);
+
+        tableSQL.add(tableTransaction);
+        tableSQL.add(tableNote);
+
+        tableSQL.add(tableTagToTrans);
+        tableSQL.add(tableTagToPayee);
+
+        try {
+            for (String statement : tableSQL) {
+                Statement stmt = database.createStatement();
+                stmt.execute(statement);
+                stmt.close();
+            }
+        } catch (SQLException e) {
+            throw new StorageException("Unable to Create Table", e);
+        }
+    }
+
+    @Override
+    public void shutdown() throws StorageException {
+        try {
+            database.close();
+        } catch (SQLException e) {
+            throw new StorageException("Exception while shutting down database.", e);
+        }
+    }
+
+    // Basic CRUD functionality
+    @Override
+    public void insertTransaction(Transaction transaction) throws StorageException {
```

Comment:
We could probably refactor this method and break out some of the pieces to make it shorter. This isn't something we need to worry about right now, however.